lib/buildFactory.js

```diff
@@ -44,9 +39,19 @@ function getCommitSha(config) {
 
         return user.unsealToken()
             .then(token => config.scm.getCommitSha({
-                scmUrl: pipeline.scmUrl,
+                scmUri: pipeline.scmUri,
                 token
-            }));
+            })
+            .then((sha) => {
+                config.modelConfig.sha = sha;
```

Comment:
What do you mean? You're just setting the config.modelConfig.sha here. You're not returning or doing anything with it in this function.